crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -388,16 +450,29 @@ struct RoomEventCacheInner {
     /// See comment there.
     store: Arc<Mutex<Arc<dyn EventCacheStore>>>,
 
+    /// A notifier that we received a new pagination token.
+    pagination_token_notifier: Notify,
```

Comment:
I wonder if we should not have a type to hold pagination data instead of 2 fields in this type.

Reply:
I think it's a bit premature at this point, but I'll keep this in mind :+1: